src/api/mod.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod error;
+mod extractors;
+mod responses;
+mod routes;
+
+use std::{
+    task::Poll,
+    time::SystemTime,
+};
+
+use actix::{
+    Actor,
+    ActorContext,
+    Context,
+    Handler,
+    Message,
+};
+use axum::{
+    http::{
+        Request,
+        Response,
+    },
+    Server,
+};
+use error::ListenerError;
+use futures::Future;
+use hyper::{
+    Method,
+    Uri,
+};
+use lazy_static::lazy_static;
+use mongodb::{
+    options::ClientOptions,
+    Client,
+};
+use pin_project::pin_project;
+use prometheus::{
+    Gauge,
+    HistogramOpts,
+    HistogramVec,
+    IntCounter,
+    IntCounterVec,
+    Opts,
+    Registry,
+};
+use routes::routes;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use tokio::sync::oneshot;
+use tower::{
+    Layer,
+    Service,
+};
+
+use crate::config::mongo::MongoConfig;
+
+/// The Chronicle API actor
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct ChronicleAPI {
+    mongo_config: MongoConfig,
```

Comment:
The API will probably be just one component of Chronicle. Should we store it as an `RwLock<Arc<MongoConfig>>` (or some other form of shared resource) here?